### DISCUSSION TITLE
Ability to set custom headers or query string parameters

### DISCUSSION THREAD

**Discussion by eithe** [GogoVega/node-red-contrib-firebase-realtime-database#73] [Ideas]
It would be great if this module would provide a way to set custom headers or querystring parameters on the in and out nodes so that you can set e.g. [writesizelimit](https://firebase.google.com/docs/reference/rest/database/#section-param-writesizelimit) when writing.

**Comment by GogoVega** [comment]
The SDK has no method to allow me to set headers for the request.

The default connection is supposed to be WebSocket, so headers are not applicable.

The docs mention also these [limits](https://firebase.google.com/docs/database/usage/limits).

**Reply by eithe** [reply]
Ah, ok, thanks for replying! I solved it for now by using the firebase CLI to set the default writeSizeLimit (`firebase database:settings:set defaultWriteSizeLimit unlimited`) which is ok for this project. Thanks for creating this module, btw, I use it for some of my side/hobby projects and it works great :)

**Reply by GogoVega** [reply]
Hi @eithe,
By rereading the discussion and according to the documentation I can define the following two settings:

- `writeSizeLimit`
- `strictTriggerValidation`

Since I am not sure of the usefulness of the second one I have only implemented the first one.

The config node has a new `Settings` tab that will allow you to define the setting without going through the CLI 😉 

<img width="649" alt="Capture d’écran 2025-02-16 à 16 53 20" src="https://github.com/user-attachments/assets/8758aaf0-3ede-4f62-bd85-3cc2b52c2331" />

(will be available in the future release)

**Reply by GogoVega** [reply]
Available now in v0.7.3 🎉 ( with some other enhancements 😉🤫 )
